pkg/providers/vsphere/internal/templates/factory.go

```diff
@@ -112,7 +113,15 @@ func (f *Factory) importOVAIfMissing(ctx context.Context, templateName, ovaURL s
 		return fmt.Errorf("failed to validate template in library for new template: %v", err)
 	}
 
-	if !templateExistsInLibrary {
+	deletedFromLibrary := false
```

Comment:
why do we need a new flag?
you could just set `templateExistsInLibrary` to false (which makes sense bc we just deleted it)